src/explore-education-statistics-admin/src/pages/admin-dashboard/components/__tests__/MethodologySummary.test.tsx

```diff
@@ -825,6 +825,102 @@ describe('MethodologySummary', () => {
     );
   });
 
+  describe('Removing an external methodology', () => {
+    test('shows the confirm modal when clicking the Remove button', async () => {
+      render(
+        <MemoryRouter>
+          <MethodologySummary
+            publication={testPublicationWithExternalMethodology}
+            topicId={testTopicId}
+            onChangePublication={noop}
+          />
+        </MemoryRouter>,
+      );
+
+      userEvent.click(
+        screen.getByTestId(
+          'Expand Details Section Ext methodolology title (External)',
+        ),
+      );
+
+      expect(
+        screen.getByRole('button', { name: 'Remove external methodology' }),
+      ).toBeInTheDocument();
+
+      userEvent.click(
+        screen.getByRole('button', { name: 'Remove external methodology' }),
+      );
+
+      await waitFor(() => {
+        expect(
+          screen.getByRole('heading', { name: 'Remove external methodology' }),
+        ).toBeInTheDocument();
+
```

Comment:
Mentioned it in Slack a few weeks back - using `getByRole` in a `waitFor` will potentially cause tests to run slower than they should do. This is due to `getByRole` being a particularly slow for larger components and when combined with `waitFor` (which basically runs things in a loop), can overload the process causing considerable slowdown or tests to even fail due to timing out.

It might not be so bad in this instance, but I would advise we avoid writing code like this so others are less likely to copy it in the future.

Instead, we should use a faster selector like `getByText` (or `getByLabel` if in a form):

```jsx
await waitFor(() => {
  expect(screen.getByText('Remove external methodology', { selector: 'h2' }))
    .toBeInTheDocument();
});

// Now do any additional assertions outside of `waitFor`
```